src/mon/OSDMonitor.cc

```diff
@@ -6152,6 +6152,13 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
       }
     } else /* var != "all" */  {
       choices_map_t::const_iterator found = ALL_CHOICES.find(var);
+      if (found == ALL_CHOICES.end()) {
+        ss << "pool '" << poolstr
+	       << "': invalid variable: '" << var << "'";
+        r = -EACCES;
```

Comment:
probably a better err would be `-EINVAL`? as `EACCES` always denotes permission denied. i understand that we are using `-EACCESS` when querying settings not applicable to certain type of pools, but i don't think we should follow them as examples.

Reply:
Done.